tests/unit/test_charm.py

```diff
@@ -85,59 +97,124 @@ def test_update_container_correct_pebble_layer(
     assert expected_plan == updated_plan
 
     service = harness.model.unit.get_container("kratos").get_service("kratos")
-    assert service.is_running()
+    assert not service.is_running()
     assert harness.model.unit.status == ActiveStatus()
 
 
-def test_cannot_connect_container(harness, mocked_kubernetes_service_patcher) -> None:
+def test_on_pebble_layer_with_database_connection(
+    harness, mocked_kubernetes_service_patcher, mocked_sql_migration
+) -> None:
+
+    harness.set_can_connect(CONTAINER_NAME, True)
+
+    initial_plan = harness.get_container_pebble_plan(CONTAINER_NAME)
+    assert initial_plan.to_yaml() == "{}\n"
+
+    setup_postgres_relation(harness)
     harness.begin()
-    harness.set_can_connect(CONTAINER_NAME, False)
 
     container = harness.model.unit.get_container(CONTAINER_NAME)
     harness.charm.on.kratos_pebble_ready.emit(container)
 
-    assert isinstance(harness.charm.unit.status, WaitingStatus)
+    expected_plan = {
+        "services": {
+            CONTAINER_NAME: {
+                "override": "replace",
+                "summary": "Kratos Operator layer",
+                "startup": "disabled",
+                "command": "kratos serve all --config /etc/config/kratos.yaml",
+            }
+        }
+    }
+    updated_plan = harness.get_container_pebble_plan(CONTAINER_NAME).to_dict()
+    assert expected_plan == updated_plan
+    updated_config = yaml.safe_load(harness.charm._config)
+    assert DB_ENDPOINTS in updated_config["dsn"]
+    assert DB_PASSWORD in updated_config["dsn"]
+    assert DB_USERNAME in updated_config["dsn"]
+
+    service = harness.model.unit.get_container("kratos").get_service("kratos")
```

Comment:
Don't validate too many things in 1 test. A unit test should only have 1 reason to fail.